### DISCUSSION TITLE
absolute save path of yarpdatadumper application

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#641] [Q&A] [answered]
The relative path of default application yarpdatadumper on yarpmanager is /data/head /data/leftArm .....  What is absolute path of save dumper data.

**Comment by pattacini** [comment]
You can specify where `yarpdatadumper` will save the recorded data by means of the `--dir` command line option.
If the option `--dir` is not specified, then the name of the directory is made up based on the name of the portname.

Check out the [documentation](https://yarp.it/latest/group__yarpdatadumper.html).

**Reply by jimingda** [reply]
I mkdir and specify the absolute save path of yarpdatadumper on yarpmanager gui (both Module "Parameters" and Connection "to" item), launch the leftArm and rightArm and connect them, the status turned green. But neither data.log or info.log appeared on the path i specified.

Shall i modify the "host" item, but i found that the /From item is correct, or modify the node item directly on xml file from localhost to icub

![2023-06-29_10-21](https://github.com/robotology/community/assets/7019692/22424d48-1cd4-4687-a1b9-2b0aa1701a45)

**Reply by pattacini** [reply, marked as answer]
Hi @jimingda 

The parameter `--name` establishes the name of the YARP port, so please stick to this and don't use `--name` in the bizarre way I can spot in your screenshot above 😃 

If you launch the following command from a terminal:
```console
yarpdatadumper --name /log --dir test
```
Then, you'll find the logged quantities in files stored under the subdirectory `./test` (relative tho the path from where you have run the command).

By contrast, if you launch `yarpdatadumper` from `yarpmanager`, files will be stored typically in the path from where you launched `yarpmanager`.

For example, if you launch `yarpmanager` from the directory `Desktop` as below, then you'll find all the relevant `yarpdatadumper` directories therein.
```console
cd ~/Desktop
yarpmanager
```

Hope it helps.

**Reply by jimingda** [reply]
Thanks. Because i launch the yarpmanager in terminal on path /home/icub,  as specified in the pic, i found that i got the yarp dump data on path below: 
/home/icub/home/icub/log/log20230629/leftArm/data.log.

And where can i find the tutorial of wholebody, except for https://robotology.github.io/robotology-documentation/doc/html/group__wholeBodyPlayer.html

**Reply by pattacini** [reply]
We don't have a proper tutorial for **`wholeBodyPlayer`**.

However, you can read more about it at https://github.com/robotology/icub-main/pull/654 and find the relative app available at https://github.com/robotology/icub-main/blob/master/app/wholeBodyPlayer/scripts/wholeBodyPlayer_App.xml.template.

> **Warning**
Be careful of the section `Pitfalls` described at https://github.com/robotology/icub-main/pull/654 that are mainly regarded with the name of the ports, which need to be fine-tuned.

> **Note**
Since this thread is concluded, if you have further questions about `wholeBodyPlayer`, please open up a brand-new discussion.
